atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraClientPoolMetricsTest.java

```diff
@@ -57,6 +58,18 @@ public void metricsAreProducedAndFiltered() {
                 .containsKey(createMeanActiveTimeMillisMetric("mean"));
     }
 
+    @Test
+    public void recordPoolSizeCorrectlySetsSize() {
+        metrics.recordPoolSize(100);
+        assertThat(metrics.getPoolSize()).isEqualTo(100);
+        metrics.recordPoolSize(-13);
+        assertThat(metrics.getPoolSize()).isEqualTo(-13);
```

Comment:
not actionable: this is fine since you're just having a gauge that reports a number, though I'd be worried if I had -13 nodes 😅

Reply:
Yes, I did that deliberately to show we won't fail hard! I'd rather a bad metric, than the cassandra client pool failing to refresh.